rpistart.sh

```diff
@@ -0,0 +1,181 @@
+
+# Create a .config file: This requires having flex and bison packages installed: 
+# Flex (Fast Lexial Analyzer Generator) is used for tokenizing input (breaking it into a series of tokens)
+# and Bison takes these tokens and analyzes their structure to understand the higher-level syntax. 
+
+ARCH=${ARCH} CROSS_COMPILE=/bin/aarch64-linux-gnu- make defconfig
+
+# Use the kvm_guest (kernel virtual machine) config as the base defconfig, which is suitable for qemu 
+ARCH=${ARCH} CROSS_COMPILE=/bin/aarch64-linux-gnu- make kvm_guest.config
+
+# Build the kernel (parallelizable if nproc > 1)
+ARCH=arm64 CROSS_COMPILE=/bin/aarch64-linux-gnu- make -j${NUM_CORES}
+
+cp arch/${ARCH}/boot/Image ${ROOT_DIR}
+
+
+# Third step: Mount the image for enabling SSH and configuring username and password and 
+# accessing the files within the image. 
+# Disk images, like those used for Raspberry Pi (e.g., RaspiOS), often contain an entire 
+# disk's worth of data, including the partition table and multiple partitions. 
+# To mount a specific partition within this image, we must calculate the correct offset where 
+# the partition starts. This calculation requires two pieces of information:
+# * sector size 
+# * starting offset 
+# For instance, if a partition starts at sector 8192 and the sector size is 512 bytes, the byte
+# offset of the partition is 8192 * 512 bytes from the start of the image.
+#
+# We have two partitions inside the generated Raspberry Pi image. The first device (partition)
+# is the bootable partition, and the second one is the root filesystem. The first partition is 
+# what will be mounted as /boot in Raspberry Pi, and this is where we'll need to create some files.
+
+
+while getopts "i:" flag
+do 
+    case "${flag}" in 
+        i)
+            IMAGE_FILE="${OPTARG}"
+            ;;
+        *)
+            ;;
+    esac
+done 
+
+function assert_numeric() {
+    if ! [[ $1 =~ ^[0-9]+$ ]] || ! [[ $1 =~ ^[0-9]+$ ]]; then
+        echo "Error: Non-numeric value detected."
+        exit 1
+    fi
+}
+
+# Check that the image exists. The image path is expected to be absolute. 
+if [ ! -f "$IMAGE_FILE" ]; then 
+    echo "Invalid Raspberry Pi OS image: '$IMAGE_FILE'"
+    exit 1
+fi 
+
+
+# Extracting sector size
+SECTOR_SIZE=$(fdisk -l $IMAGE_FILE | grep "Sector size" | awk '{print $4}')
+
+# Extracting the start of the first partition
+START_SECTOR=$(fdisk -l $IMAGE_FILE | grep -E "\.img1" | awk '{print $2}')
+
+assert_numeric $SECTOR_SIZE
+assert_numeric $START_SECTOR
+
+# Calculating the offset
+OFFSET=$(echo "$SECTOR_SIZE * $START_SECTOR" | bc)
+
+echo "Sector Size: $SECTOR_SIZE"
+echo "Start Sector: $START_SECTOR"
+echo "Offset: $OFFSET"
+
+
+# Mount the image in /mnt/rpi directory 
+sudo mkdir -p /mnt/rpi 
+if findmnt -M /mnt/rpi > /dev/null; then
+    echo "/mnt/rpi is already mounted."
+else
+    sudo mount -o loop,offset=${OFFSET} ${IMAGE_FILE} /mnt/rpi
+fi
+
+USERNAME=${USERNAME:-"pi"}
+PASSWORD=${PASSWORD:-"raspberry"}
```

Comment:
Are these defaults for the image?

Reply:
Or are we doing this to set the username/password inside the OS?

Reply:
These are for setting the username/password inside the OS. They can also be overriden through environment variables 

```shell
$ export USERNAME=<username>
$ export PASSWORD=<password>
```

I've documented this in the runner script.